core/src/main/scala/io/projectglow/transformers/LiftOverVariantsTransformer.scala

```diff
@@ -280,4 +366,51 @@ object LiftOverVariantsTransformer extends GlowLogging {
     GenerateMutableProjection.generate(
       passThroughExpressions ++ picardStructFieldLiterals :+ statusStructFieldLiteral)
   }
+
+  def getArrayFields(fields: Seq[StructField]): Seq[String] = {
+    fields.flatMap { f =>
+      if (f.dataType.isInstanceOf[ArrayType]) {
+        Some(f.name)
+      } else {
+        None
+      }
+    }
+  }
+
+  // Get array-based INFO fields
+  def getAttributesToSwap(variantSchema: StructType): Seq[String] = {
+    val infoFields = variantSchema.flatMap { f =>
+      if (f.name.startsWith(VariantSchemas.infoFieldPrefix)) {
+        Some(f.copy(name = f.name.substring(VariantSchemas.infoFieldPrefix.length)))
+      } else {
+        None
+      }
+    }
+    getArrayFields(infoFields)
+  }
+
+  // Get array-basd FORMAT fields
+  def getExtendedAttributesToSwap(variantSchema: StructType): Seq[String] = {
+    val genotypeFieldOpt = variantSchema
+      .fields
+      .find(_.name == VariantSchemas.genotypesFieldName)
+
+    if (genotypeFieldOpt.isEmpty) {
+      return Seq.empty
+    }
+
+    val genotypeField =
```

Comment:
Could we throw a better exception if `genotypes` is not an `ArrayType(StructType)`? It would be nice to put this in a common place since we frequently need to get the genotype schema.